src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+import mate.academy.repository.order.OrderRepository;
+import mate.academy.repository.orderitem.OrderItemRepository;
+import mate.academy.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.repository.user.UserRepository;
+import mate.academy.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(Long userId, String shippingAddress) {
+        ShoppingCart shoppingCart = getShoppingCart(userId);
+        User user = getUser(userId);
+
+        Order order = makeOrder(shippingAddress, user);
+        BigDecimal total = calculateTotalAndAddOrderItems(shoppingCart, order);
+
+        order.setTotal(total);
+        saveOrderAndOrderItems(order);
+
+        clearShoppingCart(shoppingCart);
+
+        return orderMapper.toDto(order);
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public List<OrderDto> getOrderHistory(Long userId, Pageable pageable) {
+        return orderRepository.findByUserId(userId, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public List<OrderItemDto> getOrderItems(Long orderId, Pageable pageable) {
+        return orderItemRepository.findByOrderId(orderId, pageable).stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public OrderItemDto getOrderItem(Long orderId, Long itemId) {
+        OrderItem orderItem = orderItemRepository.findByOrderIdAndId(orderId, itemId)
+                .orElseThrow(() -> new EntityNotFoundException("Order item not found"));
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Transactional
+    @Override
+    public void updateOrderStatus(Long orderId, Order.Status status) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found"));
+        order.setStatus(status);
+        orderRepository.save(order);
+    }
+
+    private static Order makeOrder(String shippingAddress, User user) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(shippingAddress);
+        order.setDeleted(false);
+        return order;
+    }
```

Comment:
why do you need static?

Reply:
i used "extract method" from intellij and this func makes methods static by default, i just do not noticed that.